docs/csharp/programming-guide/classes-and-structs/snippets/local-functions/Program.cs

```diff
@@ -121,5 +121,27 @@ private async Task<int> SecondStep(int index, string name)
             await Task.Delay(100);
             return 9;
         }
+
+        //<YieldReturn>
+        public IEnumerable<string> SequenceToLowercase(IEnumerable<string> input)
+        {
+            if (input.Count() == 0)
+            {
+                throw new Exception("There are no items to convert to lowercase!");
+            }
+            
+            return LowercaseIterator();
+            
+            IEnumerable<string> LowercaseIterator()
+            {
+                foreach (var item in input)
+                {
+                    var output = item.ToLower();
+                    yield return output;
```

Comment:
```suggestion
                foreach (var output in input.Select(item => item.ToLower()))
                {
                    yield return output;
```

Would require `using System.Linq;` though, so go with @pkulikov [suggestion instead](https://github.com/dotnet/docs/pull/21117/files#r505227879).

Reply:
I think I could do this since @pkulikov suggested using `if (input.Any())` instead of `if (input.Count() == 0)` for the exception above the local function.